komodo/build.py

```diff
@@ -18,30 +16,39 @@
 )
 from komodo.shell import pushd, shell
 
-flatten = itr.chain.from_iterable
 
+def full_dfs(
+    all_pkgs: Dict[str, str],
+    repo: Dict[str, Dict],
+    pkgs_to_check: Optional[List[str]] = None,
+) -> List[str]:
+    if pkgs_to_check is None:
+        pkgs_to_check = list(all_pkgs.keys())
+
+    def dfs(package_name: str, pkg_order: List[str], visited: Set[str]):
+        if package_name in visited:
+            return
+        visited.add(package_name)
+        ver = all_pkgs[package_name]
+        if not all(
+            map(all_pkgs.__contains__, repo[package_name][ver].get("depends", []))
```

Comment:
Would the code be simpler if we extracted `current_package_deps = repo[package_name][ver].get("depends", [])` , this lookup is done three places already?